neurons/validator.py

```diff
@@ -633,6 +635,12 @@ def receive_signal(self, synapse: template.protocol.SendSignal,
                 self.position_manager.save_miner_position_to_disk(open_position)
                 if miner_order_uuid:
                     self.uuid_tracker.add(miner_order_uuid)
+                # Update the last received order time
```

Comment:
this needs to be done within a lock.

Also suggest making a new class that encapsulates rate limiting, locking, holding the latest timestamp value, and writing the latest timestamp to disk. It can be a subclass of CacheController.